### ISSUE TITLE
RE: Security Vulnerability Disclosure

### ISSUE THREAD

**Issue by ian9443** [BadBoy21/N00bBot#1]
Hi BadBoy21

I have discovered what I believe to be a zero-day security vulnerability in IRCServer, while studying a computer security module at Leeds Beckett University.
I believe you were working  on the IRC Server in mid march but deleted this from github?

https://github.com/BadBoy20/IRCServer

If this is the case please continue reading, if not please ignore.
Perhaps you merged this with another IRC project you were working on? I would like to assist finding vulnerabilities with your IRCServer.

Below parts of the vulnerability policy may apply,
**Vulnerability disclosure policy**
The prefered vulnerability disclosure procedure for the security module is as follows: A student identifying a security vulnerability will make a good faith effort to privately communicate the details of the vulnerability to the vendor. The student will apply for a CVE reservation (withholding public disclosure of details, where possible). Fifteen (15) days later, the student will report the vulnerability to CERT/CC. CERT/CC will handle initial disclosure to the general public, according to their process, which typically means details are publically disclosed after a further 45 days, which gives the vendor a total of roughly 60 days to develop and release a patch. At this stage, the student may also publically release additional information or proof of concept code. 

Throughout the process the vendor/author should make an effort to provide assessment and progress report(s) to the student. Any relevant bug bounties should be proposed by the vendor to the student. The student should be given credit by name in any advisories: preferably including the text “Leeds Beckett University student”. The module leader, Dr Z. Cliffe Schreuders (c.schreuders@leedsbeckett.ac.uk), should be copied into all communications between the student and the vendor. Neither Cliffe or the university take responsibility for the student's actions (whose own responsibility it is to abide by relevant laws), but will make a good faith effort to facilitate the process of vulnerability disclosure.

The aim of this policy is to encourage responsible and coordinated disclosure, effective response, and ensures both parties are aware of the process that will be followed. 

Vulnerability description
Buffer overflows and Denial of service

System configurations believed to be vulnerable
variable ‘strArray’ was NULL.
Sending values continuously in a certain variable will always eventually cause the IRC to enter a DoS state.
Deleteing the ‘#’ from testing script
sending a random string following the USER command during the authentication phase with the server
Sending the PRIVMSG command without the room or a user identifier used after it, generates an immediate crash
Buffer oveflow array of certain variables (output10)

How the vulnerability was found
1,Backfuzz
2,Spike Fuzzer
3,Manual Testing - 10 Test cases

Impact of exploitation
User files can be edited

Student contact details
Ian Ayliffe
i.ayliffe9443@student.leedsbeckett.ac.uk
Academic contact details
Dr Z. Cliffe Schreuders
Senior Lecturer in Computer Security
Leeds Beckett University, Caedmon Hall, Headingley Campus, Leeds, LS6 3QS
+44(0)11381 28608 | c.schreuders@leedsbeckett.ac.uk | http://z.cliffe.schreuders.org

**Comment by BadBoy21** [comment]
Hi Ian.

Thank you for sharing this issue with me. To be honest I already knew that there were many vulnerabilities with that project. Which is one of the reasons I took it down. I started these projects as a hobby and I was getting to all the errors one by one by testing with valgrind as well as just looking at c coding conventions in general. Between my job as a web developer I simply did not get the chance to look at this as this was just a hobby of mine.

I also am aware there are probably similar errors in 'N00bbot' as well which is why I probably will move the project to a developer branch and with in it there. Actually when I started these projects I did it mostly to learn and experiment with c. And learn git as well.

Anyway, I really appreciate the feed back and welcome it. Feel free to raise more issues about this project as you see them and I will tend to them when I can. Thanks. And with that said, since this is most topical to Ircserver (which I have backed up away somewhere and will put it back up when I get the chance) I will close this issue.

Yours,
Jessey
